chapter2/tests/t1/chapter2-t1-two-files.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -eo pipefail
+# Args check
+if [[ "$#" -ne 2 ]]; then
+    echo "Error: Two arguments are required"
+    exit 1
+fi
+# Function for delimiter
+print_separator() {
+    printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -
+}
+# Function to generate a random string and write to a file with permissions
+gen_rand_str() {
+    local file="$1"
+    openssl rand -base64 20 > "$file" && chmod 700 "$file"
+}
+
+if   [[ ! -r `pwd` ]]; then
+    echo "Directory is not readable."
+    exit 255
+elif [[ -e $1 && -e $2 ]]; then
+    print_separator
+    echo file $1 is here: && cat $1
```

Comment:
You still have a lot of similar lines (DRY). Please reorganize to loop and see how better your code will be

Reply:
> You still have a lot of similar lines (DRY). Please reorganize to loop and see how better your code will be
Thanks, seems to have fixed everything